src/tls/cockpit-certificate-ensure.c

```diff
@@ -56,6 +56,9 @@
 // We tolerate the deprecated merged cert/key files only for cockpit-tls.
 static bool tolerate_merged_cert_key;
 
+// Depending on the mode, some errors are recoverable.
+static bool exit_on_error;
```

Comment:
Can you please describe what this should achieve? My first reaction to this is "really really no" --- certificate-ensure should *never* exit with 0 if the certificate is invalid or missing, with or without the `--check` option. The purpose of `--check` is not to enable error reporting, but to disable the actual creation of a certificate if it's missing (think of `--no-act` or `--dry-run`).

Reply:
You are right, I should write `For cockpit-tls, some errors are recoverable.`.

This PR must not change the behavior of other use cases (`--check` or no options).